### PR TITLE
Bug 1872663: Gherkin Script Design for Epic ODC-4230

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4501

Problem: Gherkin Scripts Design for Connector Event Source epic ODC-4230

Solution: These are the gherkin scripts designed covering all the scenarios related to Connector Event Source epic ODC-4230

### CODE REVIEW

**Comment by gajanan-more** [comment]
/retitle Bug 1872663: Gherkin Script Design for Epic ODC-4230

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    And user will see AWS Kinesis card
+    And user will see Jira card
+    And user will see Slack card
+    And user will see Telegram card
+
+
+@regression
+Scenario: Create AWS 2 Simple Queue Service connector
+    Given user is at Event Sources page
+    When user click on AWS Sqs card
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:30]: *clicks

Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature]: updated

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature [hunk 2]
```diff
@@ -0,0 +1,78 @@
+Feature: Connector Event Sources
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:1]: would be good to use `CamelK connector Event Sources` for more clarity as these are CamelK connectors

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    User should be able to install Connector Event Sources
+
+
+#Scenarios of installing these operator has been included in operators.feature file
+Background: 
+    Given user has installed the OpenShift Serverless Operator
+    And user has created knative serving
+    And user has create knative eventing
+    And user has installed Knative Apache Camel Operator
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:10]: IMO name of operator is just `knative camel Operator`

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    And user will see AWS Kinesis card
+    And user will see Jira card
+    And user will see Slack card
+    And user will see Telegram card
+
+
+@regression
+Scenario: Create AWS 2 Simple Queue Service connector
+    Given user is at Event Sources page
+    When user clicks on AWS Sqs card
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:30]: IMO can add more details, WDYT?
```suggestion
    When user clicks on AWS Sqs card, user would be shown with Yaml Editor and user update/modify spec as needed
```

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:10]: No. It's Knative Apache Camel Operator.

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:1]: Updated.

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:30]: I will add a separate scenario for this

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    Then user will be redirected to Topology page
+    And user will see Telegram connector
+
+
+@regression, @manual
+Scenario: Update YAML in the editor while installing CamelK Connector Event Source
+    Given user is at Event Sources page
+    When user clicks on Telegram card
+    And user sees the YAML editor
+    And user updates the metadata.name field in the YAML to TelegramConnector
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:86]: user can update anything values in spec, FYI most of these connectors needs auth token , which user may have to provide if they want to see actual events `authorizationToken: <put-here-the-token-from-the-bot-father>`

File: frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    And user has installed Knative Apache Camel Operator
+    And user is at Developer perspective
+
+
+@regression
+Scenario: CamelK Connector Event Source cards
+    Given user is at Developer perspective
+    When user goes to +Add page
+    And user clicks on the Event Sources card
+    Then user will see the AWS 2 Simple Queue Service card
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:19]: Will the cards are the same you mentioned or may change

Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/connector-Event-Sources-ODC-4230.feature:19]: No. As of now, there won't be any change.

**Comment by sanketpathak** [comment]
> Clicking on one of these event sources In topology view should show a side panel, including a Resources tab which has the associated resources
Are we not targeting this as it is mentioned in the epic

**Pull request merged**